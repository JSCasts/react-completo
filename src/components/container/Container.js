import React from 'react';
import { Link } from 'react-router';

import logo from '../../assets/images/dribble-logo.png';
import styles from './styles.scss';

const Container = ({ children }) => (
  <div>
    <header className={styles.header}>
      <Link to="/"><img src={logo} alt="Logo" /></Link>
    </header>
    <main className={styles.main}>
      {children}
    </main>
  </div>
);

Container.propTypes = {
  children: React.PropTypes.element.isRequired,
};

export default Container;
