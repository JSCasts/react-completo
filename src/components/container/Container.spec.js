import React from 'react';
import renderer from 'react-test-renderer';
import { Router } from 'react-router';

import Container from './Container';

const children = React.createElement('div');

test('Container', () => {
  const component = renderer.create(
    <Container children={children} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
