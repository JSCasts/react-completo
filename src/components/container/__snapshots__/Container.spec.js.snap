exports[`test Container 1`] = `
<div>
  <header
    className="header">
    <a
      onClick={[Function]}
      style={Object {}}>
      <img
        alt="Logo"
        src="test-file-stub" />
    </a>
  </header>
  <main
    className="main">
    <div />
  </main>
</div>
`;
